compiler/rustc_borrowck/src/region_infer/opaque_types.rs

```diff
@@ -586,6 +587,21 @@ impl<'tcx> TypeFolder<'tcx> for ReverseMapper<'tcx> {
                 self.tcx.mk_generator(def_id, substs, movability)
             }
 
+            ty::GeneratorWitnessMIR(def_id, substs) => {
```

Comment:
Could this be merged with the `ty::Closure` match arm (with a check at the end to call `self.tcx.mk_closure(def_id, substs)` vs `self.tcx.mk_generator_witness_mir(def_id, substs)`? I assume the reasoning is the same (applied to 'the closure part' of the generator), and there's a more detailed comment there.